src/NHapi.Base/Parser/ParserBase.cs

```diff
@@ -121,13 +112,13 @@ public static ISegment MakeControlMSH(string version, IModelClassFactory factory
                     (IMessage)GenericMessage
                         .GetGenericMessageClass(version)
                         .GetConstructor(new[] { typeof(IModelClassFactory) })
-                        .Invoke(new object[] { factory });
+                        ?.Invoke(new object[] { factory });
```

Comment:
this would have thrown previously

Reply:
You are correct, will review if this makes a difference.

Reply:
same here will change to `!.` as known not to be null.